sql/core/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowUtils.scala

```diff
@@ -42,6 +43,13 @@ object ArrowUtils {
     case StringType => ArrowType.Utf8.INSTANCE
     case BinaryType => ArrowType.Binary.INSTANCE
     case DecimalType.Fixed(precision, scale) => new ArrowType.Decimal(precision, scale)
+    case DateType => new ArrowType.Date(DateUnit.DAY)
+    case TimestampType =>
+      timeZoneId match {
+        case Some(id) => new ArrowType.Timestamp(TimeUnit.MICROSECOND, id)
```

Comment:
Here we are mapping Spark SQL timestamp type to Arrow timestamp with timezone type. Does Arrow have tz-naive timestamp type?

Reply:
Looking at the read size, the timezone id here is not used? We always convert to python local timezone.

Reply:
I think we should use UTC here, then at read path, after we convert arrow to pandas, we should update pandas dataframe's timezone, to python local timezone.

Reply:
After we agree `toPandas` has a wrong behavior, we can update pandas dataframe's timezone to session local timezone in the last step.

Reply:
ok I understand the reason now. Setting the timezone here is better than setting it at python side. My only question is, why the timezone id is optional?